ponyo/pipeline.py

```diff
@@ -104,9 +104,29 @@ def setup_dir(config_file):
             os.makedirs(each_dir, exist_ok=True)
 
 
-def normalize_expression_data(base_dir, config_file, raw_input_data_file):
+def transpose_data(data_file, out_file):
     """
-    Normalize the expression data
+    Transpose and save expression data so that it is of the form sample x gene
+
+    Arguments
+    ----------
+    data_file: str
+        File containing gene expression
+
+    out_file: str
+        File containing transposed gene expression
+    """
```

Comment:
Not a big deal if you don't plan on expanding these functions a lot, but typically it's best to have each function have a single purpose, such as taking the transpose of the data or writing it to a file. As things are, if you want the transpose of a matrix created by this function you have to write it to disk and read in from disk instead of just getting it as a return.

I imagine that this isn't a problem for your current workflow, but it's something to think about in the future.

Reply:
I see your point. That makes sense. In this case the reason to write it to file was so that this transposed matrix file could serve as the input into the rest of the pipeline so I didn't have to keep transposing it each time I needed the data.